Test/Unit/Model/Api/ShippingMethodsTest.php

```diff
@@ -346,84 +366,165 @@ public function getShippingMethodsWithFullAddressData()
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->configHelper->method('getStoreVersion')->willReturn('2.3.3');
+
+        $methods = ['sendErrorResponse', 'checkCartItems', 'getQuoteById',
+            'shippingEstimation', 'preprocessHook', 'couponInvalidForShippingAddress'
         ];
-        $this->currentMock = $this->getMockBuilder(BoltShippingMethods::class)
-            ->setMethods($methods)
-            ->setConstructorArgs([
-                $this->hookHelper,
-                $this->regionModel,
-                $this->factoryShippingOptionsMock,
-                $this->shippingTaxInterfaceFactory,
-                $this->cartHelper,
-                $this->totalsCollector,
-                $this->converter,
-                $this->shippingOptionInterfaceFactory,
-                $this->bugsnag,
-                $this->metricsClient,
-                $this->logHelper,
-                $this->errorResponse,
-                $this->response,
-                $this->configHelper,
-                $this->request,
-                $this->cache,
-                $this->priceHelper,
-                $this->sessionHelper,
-                $this->discountHelper,
-                $this->ruleFactory
-            ])
+
+        $this->sessionHelper->expects(self::once())->method('loadSession')->willReturn(null);
+
+        $this->initCurrentMock($methods, false);
+
+        $this->currentMock->expects(self::once())->method('preprocessHook')->willReturn(null);
+        $this->currentMock->expects(self::once())->method('checkCartItems')->with($cart)->willReturn(null);
+
+        $this->currentMock->expects(self::exactly(2))->method('getQuoteById')
+            ->withConsecutive([self::IMMUTABLE_QUOTE_ID], [self::PARENT_QUOTE_ID])
+            ->willReturnOnConsecutiveCalls($quote, $quote);
+
+        $shippingOptions = $this->getShippingOptions();
+
+        $this->currentMock->method('shippingEstimation')
+            ->willReturn($shippingOptions);
+
+        $this->currentMock->expects(self::once())->method('couponInvalidForShippingAddress')
+            ->withAnyParameters()->willReturn(false);
+
+        $result = $this->currentMock->getShippingMethods($cart, $shippingAddress);
+
+        $this->assertEquals($shippingOptions, $result);
+    }
+
+    /**
+     * @test
+     */
+    public function getShippingMethods_taxAdjustedAndInvalidCoupon()
+    {
+        $cart = [
+            'display_id'      => self::DISPLAY_ID,
+            'order_reference' => self::PARENT_QUOTE_ID
+        ];
+        $shippingAddress = [
+            'company'         => "",
+            'country'         => "United States",
+            'country_code'    => "US",
+            'email'           => "integration@bolt.com",
+            'first_name'      => "YevhenBolt",
+            'last_name'       => "BoltTest2",
+            'locality'        => "New York",
+            'phone'           => "+1 231 231 1234",
```

Comment:
@matt-thomason @danac-gs 
As i see last time Bolt have another format of phone.
It should be like `2312311234` without `+1` code

Reply:
done